File: api/src/test/java/net/blueshell/api/factory/model/TelemetryFactory.java
package net.blueshell.api.factory.model;

import net.blueshell.api.testutil.ModelTestUtils;
import com.github.javafaker.Faker;
import lombok.RequiredArgsConstructor;
import net.blueshell.api.common.enums.PlatformType;
import net.blueshell.api.model.Telemetry;
import org.springframework.stereotype.Component;

import java.util.concurrent.atomic.AtomicLong;

/**
 * Factory for Telemetry model test instances.
 */
@Component
@RequiredArgsConstructor
public class TelemetryFactory {

    private static final AtomicLong counter = new AtomicLong(1000);
    private final Faker faker;

    public Telemetry createBasic() {
        Telemetry t = new Telemetry();
        ModelTestUtils.setId(t, generateId());
        t.setPlatform(faker.options().option(PlatformType.class));
        t.setUrl(faker.internet().url());
        return t;
    }

    public Telemetry createFull() {
        return createBasic();
    }

    public Telemetry createWithCustomizations(java.util.function.Consumer<Telemetry> customizer) {
        Telemetry t = createFull();
        customizer.accept(t);
        return t;
    }

    private Long generateId() {
        return counter.incrementAndGet();
    }
}

File: api/test/old/api/factory/model/TelemetryFactory.kt
package net.blueshell.api.factory.model

import com.github.javafaker.Faker
import net.blueshell.api.domain.telemetry.persistence.Telemetry
import net.blueshell.api.shared.enums.PlatformType
import org.springframework.stereotype.Component
import java.util.concurrent.atomic.AtomicLong
import java.util.function.Consumer

/**
 * Factory for Telemetry model test instances.
 */
@Component
class TelemetryFactory(
    private val faker: Faker
) {

    fun createBasic(): Telemetry {
        val telemetry = Telemetry()
        telemetry.platform = faker.options().option(PlatformType::class.java)
        telemetry.url = faker.internet().url()
        return telemetry
    }

    fun createFull(): Telemetry = createBasic()

    fun createWithCustomizations(customizer: Consumer<Telemetry>): Telemetry {
        val telemetry = createFull()
        customizer.accept(telemetry)
        return telemetry
    }

    private fun generateId(): Long = COUNTER.incrementAndGet()

    private companion object {
        val COUNTER = AtomicLong(1000)
    }
}
